docs/apiref/firmware.rst

```diff
@@ -132,3 +136,88 @@ To remove a firmware image:
     "job_id": "5d849177dd428720db72c693"
    }
     
+
+Upgrade firmware on device(s)
+-----------------------------
+
+As of today we support upgrading firmware on Arista EOS acces
+switches. The upgrade procedure can do a 'pre-flight check' which will
+make sure there is enough disk space before attempting to download the
+firmware.
+
+The API method will accept a few parameters:
+
+* group: The name of a group, all devices in that group will be upgraded.
+* hostname: If a hostname is specified, this single device will be upgraded.
+* filename: Name of the new firmware, for example "test.swi".
+* url: URL to the firmware storage, for example "http://hostname/firmware/". This should typically point to the CNaaS NMS server and files will be downloaded from the CNaaS HTTP server.
+* download: Only download the firmware.
+* pre_flight: If true, check disk-space etc before downloading the firmware.
+* activate: Control whether we should install the new firmware or not.
+* reboot: When the firmware is downloaded, reboot the switch.
+* start_at: Schedule a firmware upgrade to be started sometime in the future.
+
+An example CURL command can look like this:
+::
+
+   curl -k -s -H "Content-Type: application/json" -X POST https://hostname/api/v1.0/upgrade -d '{"group": "ACCESS", "filename": "test_firmware.swi", "url": "http://hostname/", "pre-flight": true, "download": true, "activate": true, "reboot": true, "start_at": "2019-12-24 00:00:00"}'
+
+The output from the job will look like this:
+
+::
+
+  {
+    "status": "success",
+    "data": {
+      "jobs": [
+        {
+          "id": "5dcd110a5670fd67a615b089",
+          "start_time": "2019-11-14 08:32:11.135000",
+          "finish_time": "2019-11-14 08:34:50.352000",
+          "status": "FINISHED",
+          "function_name": "device_upgrade",
+          "result": {
+            "eoaccess": [
+              {
+                "name": "device_upgrade_task",
+                "result": "Pre-flight check done.",
+                "diff": "",
+                "failed": false
+              },
+              {
+                "name": "arista_pre_flight_check",
+                "result": "Firmware download done.",
```

Comment:
result says firmware download done, but it should just be a pre flight check?

Reply:
Fixed.